samples/net/sockets/tls_configurations/CMakeLists.txt

```diff
@@ -0,0 +1,32 @@
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.20.0)
+
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(tls_configurations)
+
+target_sources(app PRIVATE src/main.c)
+
+# A couple of helper functions to convert a certificate input file in PEM format
```

Comment:
I saw `generate_inc_file_for_target()` in the `echo_client` sample application (the one I started from), but it was not doing what I needed here.  In that sample `generate_inc_file_for_target` is used to convert a DER file (binary) to a C-array-like format. When I tried to do the same here, I got CI complaining that I could not include binary files in Zephyr so that's why I ended up making my own version of `generate_struct_content` + `scripts/convert_pem_cert.py`.

However now that you made me think about this again I realized that perhaps the first time I tried the CI complained because the file did not have the `*.der` extension for the certificate. I will give this a try to see if I can make this work with already existing functions (I agree it would be better).

Reply:
Using DER certificate format will still give you a warning, even with .der extension (which IMO is a bit dumb, but ok). My point was, that instead of converting the cert in PEM format manually, we could simply convert the certificate in to C array, and let mbed TLS deal with PEM cert (unless there are valid reasons not to do so).

Reply:
Oh, that's indeed a nice alternative. I will give it a try